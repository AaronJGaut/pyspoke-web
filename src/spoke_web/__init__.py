from . import conn
from .version import __version__

