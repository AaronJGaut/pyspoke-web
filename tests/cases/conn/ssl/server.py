import os
import asyncio
import spoke_web


cert = "../../../common/localhost.pem"
clients = []


async def handle_client(conn):
    clients.append(conn)
    print("server.handle_connect")
    try:
        async for msg in conn:
            print(f"server.handle_recv: {msg}")
            for client in clients:
                await client.send(msg)
    except ConnectionError:
        pass
    finally:
        print("server.handle_disconnect")
        clients.remove(conn)


async def main():
    async with spoke_web.conn.Server(cert=cert) as server:
        async for client in server:
            asyncio.create_task(handle_client(client))


try:
    asyncio.run(main())
except KeyboardInterrupt:
    pass
