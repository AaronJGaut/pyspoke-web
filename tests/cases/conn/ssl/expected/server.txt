server.handle_connect
server.handle_recv: b'client1 0'
server.handle_recv: b'client1 1'
server.handle_recv: b'client1 2'
server.handle_connect
server.handle_recv: b'client2 0'
server.handle_recv: b'client1 3'
server.handle_recv: b'client2 1'
server.handle_recv: b'client1 4'
server.handle_recv: b'client2 2'
server.handle_disconnect
server.handle_recv: b'client2 3'
server.handle_recv: b'client2 4'
server.handle_disconnect
